docs/design/autoware-interfaces/components/localization.md

```diff
@@ -0,0 +1,149 @@
+
+Environment map created with point cloud, published by the map server.
+
+- sensor_msgs/msg/PointCloud2
+
+A 3d point cloud map is used for LiDAR-based localization in Autoware.
+
+### Manual Initial Pose
+
+Start pose of ego, published by the user interface.
+
+- geometry_msgs/msg/PoseWithCovarianceStamped
+  - std_msgs/msg/Header header
+  - geometry_msgs/msg/PoseWithCovariance pose
+    - geometry_msgs/msg/Pose pose
+      - geometry_msgs/msg/Point position
+      - geometry_msg/msg/Quaternion orientation
+    - double[36] covariance
+
+### 3D-LiDAR Scanning
+
+LiDAR scanning for NDT matching, published by the LiDAR sensor.
+
+- sensor_msgs/msg/PointCloud2
+
+The raw 3D-LiDAR data needs to be processed by the [point cloud pre-processing modules](../../autoware-architecture/sensing/data-types/point-cloud.md) before being used for localization.
+
+### Automatic Initial pose
+
+Start pose of ego, calculated from INS(Inertial navigation sensor) sensing data.
+
+- geometry_msgs/msg/PoseWithCovarianceStamped
+  - std_msgs/msg/Header header
+  - geometry_msgs/msg/PoseWithCovariance pose
+    - geometry_msgs/msg/Pose pose
+      - geometry_msgs/msg/Point position
+      - geometry_msg/msg/Quaternion orientation
+    - double[36] covariance
+
+When the initial pose is not set manually, the message can be used for automatic pose initialization.
+
+Current Geographic coordinate of the ego, published by the GNSS sensor.
+
+- sensor_msgs/msg/NavSatFix
+  - std_msgs/msg/Header header
+  - sensor_msgs/msg/NavSatStatus status
+  - double latitude
+  - double longitude
+  - double altitude
+  - double[9] position_covariance
+  - unit8 position_covariance_type
+
+Current orientation of the ego, published by the IMU.
+
+- autoware_sensing_msgs/msg/GnssinsOrientationStamped
+  - std_msgs/Header header
+  - autoware_sensing_msgs/msg/GnssinsOrientation orientation
+    - geometry_msgs/Quaternion orientation
+    - float32 rmse_rotation_x
+    - float32 rmse_rotation_y
+    - float32 rmse_rotation_z
+
+### IMU Data
+
+Current orientation, angular velocity and linear acceleration of ego, calculated from IMU sensing data.
+
+- sensor_msgs/msg/Imu
+  - std_msgs/msg/Header header
+  - geometry_msgs/msg/Quaternion orientation
+  - double[9] orientation_covariance
+  - geometry_msgs/msg/Vector3 angular_velocity
+  - double[9] angular_velocity_covariance
+  - geometry_msgs/msg/Vector3 linear_acceleration
+  - double[9] linear_acceleration_covariance
+
+### Vehicle Velocity Status
+
+Current velocity of the ego vehicle, published by the vehicle interface.
+
+- autoware_auto_vehicle_msgs/msg/VelocityReport
+  - std_msgs/msg/Header header;
+  - float longitudinal_velocity;
+  - float lateral_velocity;
+  - float heading_rate;
+
+Before the velocity input localization interface, module `vehicle_velocity_converter` converts message type `autoware_auto_vehicle_msgs/msg/VelocityReport` to `geometry_msgs/msg/TwistWithCovarianceStamped`.
+
+## Outputs
+
+### Vehicle pose
```

Comment:
Position and velocity are now included in kinematic_state and are no longer output as localization output.

Reply:
if we also need pose, velocity and acceleration, we can use
PoseWithCovarianceStamped,TwistWithCovarianceStamped and AccelWithCovarianceStamped.

I don't know what will happen to the code in the future, but the localizationtion module of the current code does output these topics at the same time.

Reply:
@cyn-liu sorry, I thought I had deleted the `/localization/pose_with_covariance` topic, but I see it is still there.

If that is the case, you can write Pose as output for documentation.
However, it would be helpful if you could specify that it is deprecated to use it as an output for localization, as it may be deleted in the future.

Reply:
@YamatoAndo  Thanks, your suggestion is very helpful to me.  I will specify in the document that the pose and twist topics have been included in the kinematics_ State, and will be abandoned in the future.

Reply:
@YamatoAndo  hello, I have a question:
The data type of the currently localization output `kinematic_state` topic is `nav_msgs/msg/Odometry`, only includes `pose` and `twist`, however,  the custom [KinematicState](https://github.com/autowarefoundation/autoware_msgs/blob/main/autoware_localization_msgs/msg/KinematicState.msg) includes 'pose', `twist` and `acceleration`, but is not used.

Reply:
@cyn-liu 
You are correct, it is currently nav_msgs/msg/Odometry.
but will be changed to the custom [KinematicState](https://github.com/autowarefoundation/autoware_msgs/blob/main/autoware_localization_msgs/msg/KinematicState.msg) in the future.

I have created  a issue https://github.com/autowarefoundation/autoware.universe/issues/3024.

Reply:
OK, thanks @YamatoAndo 
I will update to custom [KinematicState](https://github.com/autowarefoundation/autoware_msgs/blob/main/autoware_localization_msgs/msg/KinematicState.msg) in this documentation.